### PR TITLE
doc: add example for Stdin::read_line

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/libstd/io/stdio.rs [hunk 1]
```diff
@@ -204,6 +204,30 @@ impl Stdin {
     ///
     /// For detailed semantics of this method, see the documentation on
     /// `BufRead::read_line`.
+    ///
+    /// # Examples
+    ///
+    /// ```
```
Comment (alexcrichton) [inline src/libstd/io/stdio.rs]: This should be tagged `no_run` as we probably don't want to run it during doctests.

File: src/libstd/io/stdio.rs [hunk 2]
```diff
@@ -204,6 +204,30 @@ impl Stdin {
...
+    /// use std::io;
+    ///
+    /// #fn main() {
```
Comment (alexcrichton) [inline src/libstd/io/stdio.rs]: This `fn main()` wrapper should be added automatically so there's no need to have this here or the indentation of the example.

File: src/libstd/io/stdio.rs [hunk 3]
```diff
@@ -204,6 +204,30 @@ impl Stdin {
...
+    ///     let mut input = String::new();
+    ///     match io::stdin().read_line(&mut input) {
+    ///         Ok(n)      => {
+    ///             println!("{} characters read", n);
```
Comment (alexcrichton) [inline src/libstd/io/stdio.rs]: Idiomatically we don't tend to put the extra spaces before the `=>` character, and the return value here is the number of bytes read, not the number of characters read.

Comment (alexcrichton) [inline src/libstd/io/stdio.rs]: I believe that you'll also need to import `std::io::prelude::*` to get this example to work. (you can be sure to run this test via `make check-stage2-doc-crate-std`).

Comment (alexcrichton) [inline src/libstd/io/stdio.rs]: I believe that you'll also need to import `std::io::prelude::*` to get this example to work. (you can be sure to run this test via `make check-stage2-doc-crate-std`).

Comment (tshepang) [inline src/libstd/io/stdio.rs]: The example works as-is... no further imports needed

Comment (tshepang) [inline src/libstd/io/stdio.rs]: The output there will read something like:

> 3 characters read    
> foo

File: src/libstd/io/stdio.rs [hunk 4]
```diff
@@ -204,6 +204,28 @@ impl Stdin {
...
+    ///
+    /// let mut input = String::new();
+    /// match io::stdin().read_line(&mut input) {
+    ///     Ok(n) => {
+    ///         println!("{} bytes read", n);
+    ///         println!("{}", input);
+    ///     }
+    ///     Err(error) => println!("error: {}", error),
+    /// }
+    ///```
```
Comment (steveklabnik) [inline src/libstd/io/stdio.rs]: need a space between `///` and the backticks

Comment (tshepang) [inline src/libstd/io/stdio.rs]: wow, so unforgiving :)

Comment (tshepang) [inline src/libstd/io/stdio.rs]: fixed

**Pull request merged**